### PR TITLE
Support DID Ion Recovery and Deactivate 

### LINKED ISSUE #3: Implement did:ion Deactivate Operation
**Background/context**: [click here](https://github.com/TBD54566975/web5-js/issues/181).

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
This PR introduces `recover` and `deactivate` for ion DIDs. Fixes #3 and #2.

# Description
Some notes in this PR:
* The `recover` operation API design was so that it matches the `create` API.
* Operations now return an `Ion<operation_type>Result` object. This makes working with DIDs easier.
* Examples were added in the `examples.md` readme.
* Small fixes to ensure the issuance also works with an ION did that was created with an AWS key manager.
* The changes in #87 are merged, as they're needed for tests to pass.
* It's built on top of #77 

# How Has This Been Tested?
Unit tests similar to the ones in the ion-sdk were added.

## References
* https://identity.foundation/sidetree/api/#recover
* https://identity.foundation/sidetree/api/#deactivate

### CODE REVIEW

File: credentials/src/main/kotlin/web5/sdk/credentials/VerifiableCredential.kt [hunk 1]
```diff
@@ -87,9 +92,14 @@ public class VerifiableCredential(public val vcDataModel: VcDataModel) {
     val algorithm = publicKeyJwk.algorithm
     val jwsAlgorithm = JWSAlgorithm.parse(algorithm.toString())
 
+    val kid = when (assertionMethod.id.isAbsolute) {
```
Comment (nitro-neal) [inline credentials/src/main/kotlin/web5/sdk/credentials/VerifiableCredential.kt:95]: fancy if statement yea?

File: dids/src/test/kotlin/web5/sdk/dids/DidIonTest.kt [hunk 2]
```diff
@@ -41,6 +44,19 @@ class DidIonTest {
     assertTrue(did.creationMetadata!!.longFormDid.startsWith(did.uri))
   }
 
+  @Test
+  @Ignore("For demonstration purposes only - relies on network and AWS configuration")
```
Comment (nitro-neal) [inline dids/src/test/kotlin/web5/sdk/dids/DidIonTest.kt:48]: cool, eventually we should add a lot of these in another demonstration file maybe

**Review by nitro-neal — APPROVED** [review]
awesome!
Does this need to be tested in real life with an onchain transaction?

File: credentials/src/main/kotlin/web5/sdk/credentials/VerifiableCredential.kt (hunk 1 above)
Comment (andresuribe87) [reply credentials/src/main/kotlin/web5/sdk/credentials/VerifiableCredential.kt:95]: Yeah

File: dids/src/test/kotlin/web5/sdk/dids/DidIonTest.kt (hunk 2 above)
Comment (andresuribe87) [reply dids/src/test/kotlin/web5/sdk/dids/DidIonTest.kt:48]: Definitely! I tried to do a bit of that in the `examples.md` file, but wanted to have a test file to at least ensure the syntax is valid.

**Comment by andresuribe87** [comment]
> awesome! Does this need to be tested in real life with an onchain transaction?

@nitro-neal yes, we definitely should!

**Pull request merged**